app/controllers/katello/api/v2/host_packages_controller.rb

```diff
@@ -14,66 +14,74 @@ class Api::V2::SystemPackagesController < Api::V2::ApiController
       end
     end
 
-    api :PUT, "/systems/:system_id/packages/upgrade", N_("Update packages remotely"), :deprecated => true
-    param :system_id, :identifier, :required => true, :desc => N_("UUID of the content-host")
+    api :PUT, "/hosts/:host_id/packages/upgrade", N_("Update packages remotely")
+    param :host_id, :identifier, :required => true, :desc => N_("ID of the host")
     param :packages, Array, :desc => N_("list of packages names"), :required => true
     def upgrade
       if params[:packages]
         packages = validate_package_list_format(params[:packages])
-        task     = async_task(::Actions::Katello::System::Package::Update, @system, packages)
+        task     = async_task(::Actions::Katello::Host::Package::Update, @host, packages)
         respond_for_async :resource => task
       end
     end
 
-    api :PUT, "/systems/:system_id/packages/upgrade_all", N_("Update packages remotely"), :deprecated => true
-    param :system_id, :identifier, :required => true, :desc => N_("UUID of the content-host")
+    api :PUT, "/hosts/:host_id/packages/upgrade_all", N_("Update packages remotely")
+    param :host_id, :identifier, :required => true, :desc => N_("ID of the host")
     def upgrade_all
-      task     = async_task(::Actions::Katello::System::Package::Update, @system, [])
+      task     = async_task(::Actions::Katello::Host::Package::Update, @host, [])
       respond_for_async :resource => task
     end
 
-    api :PUT, "/systems/:system_id/packages/remove", N_("Uninstall packages remotely"), :deprecated => true
-    param :system_id, :identifier, :required => true, :desc => N_("UUID of the content-host")
+    api :PUT, "/hosts/:host_id/packages/remove", N_("Uninstall packages remotely")
+    param :host_id, :identifier, :required => true, :desc => N_("ID of the host")
     param_group :packages_or_groups
     def remove
       if params[:packages]
         packages = validate_package_list_format(params[:packages])
-        task     = async_task(::Actions::Katello::System::Package::Remove, @system, packages)
+        task     = async_task(::Actions::Katello::Host::Package::Remove, @host, packages)
         respond_for_async :resource => task
         return
       end
 
       if params[:groups]
         groups = extract_group_names(params[:groups])
-        task   = async_task(::Actions::Katello::System::PackageGroup::Remove, @system, groups)
+        task   = async_task(::Actions::Katello::Host::PackageGroup::Remove, @host, groups)
         respond_for_async :resource => task
       end
     end
 
     private
 
-    def find_system
-      @system = System.where(:uuid => params[:system_id]).first
-      fail HttpErrors::NotFound, _("Couldn't find system '%s'") % params[:system_id] if @system.nil?
-      @system
+    def find_host
+      @host = ::Host::Managed.find(params[:host_id])
+      fail HttpErrors::NotFound, _("Couldn't find host with host id '%s'") % params[:host_id] if @host.nil?
+      @host
     end
```

Comment:
I would add a check here to verify the host has a content facet and a uuid in that content facet (implying its registered via pulp)